### PR TITLE
DOC: Correct usage example for `np.char.decode` docstring

### LINKED ISSUE #22015: DOC: The example for `np.char.decode` gives an example of `encode`
### Issue with current documentation:

The example in the docstring of [`np.char.decode`](https://numpy.org/doc/stable/reference/generated/numpy.char.decode.html) shows the use of `encode`, not `decode`:

https://github.com/numpy/numpy/blob/5ba36b700284a8392af5be542f65ecd262fd2568/numpy/core/defchararray.py#L565-L568

This should be fixed to demonstrate `decode`.

### PR DESCRIPTION
**Pull request by BrentTanKian**

### CODE REVIEW

**Review by mattip — COMMENTED** [review]
This looks fine and fixes what looks like a copy-paste error. If you have more energy, the `See Also` link on line 554 is broken. I think it should be 
```
:py:meth:`str.decode`
```

**Comment by BrentTanKian** [comment]
Hey sure, thanks for the heads up @mattip. I'll take a look and push up the new changes

**Comment by mattip** [comment]
Nope, the [incantation was wrong](https://app.circleci.com/pipelines/github/numpy/numpy/15223/workflows/20bee806-1166-4df0-9545-774cb70b50ca/jobs/27332?invite=true#step-108-233). I misled you, the method is `bytes.decode` (there is no `str.decode`).

It seems you also picked up a stray change in a different file, you might need to rebase and force-push or merge main back into the branch

**Comment by BrentTanKian** [comment]
I hope that did the trick. Rebased with the upstream main branch and force pushed, and removed the recent changes

**Comment by mattip** [comment]
I took the liberty of adding a change of `str.decode` -> `bytes.decode`.

**Comment by melissawm** [comment]
Hi @BrentTanKian - was this closed by accident? Feel free to reopen if that's the case, and let us know if you need help moving forward. Thanks!

**Comment by BrentTanKian** [comment]
Hmm I'm not sure. The changes seem to have been finalized and vetted through but they haven't been merged with the master branch yet. Perhaps you could review the changes again, and if it looks good to you, you could merge the changes into master. And I also don't think I have the permissions to reopen the issue

**Comment by WarrenWeckesser** [comment]
Check out the rendered docstring at [numpy.char.decode](https://output.circle-artifacts.com/output/job/599a9f4b-0f68-4564-b718-ff8327c1261d/artifacts/0/doc/build/html/reference/generated/numpy.char.decode.html#numpy.char.decode).

The code shown in the rendered doc is not valid Python, because `b'ÁÁÁ'` will generate a `SyntaxError`.   Making the entire docstring a raw string by prefixing it with `r`  at https://github.com/numpy/numpy/blob/abbf86b4a2f7692ab74a6c395c4e0a7f953a868d/numpy/core/defchararray.py#L531 should fix that.

File: numpy/core/defchararray.py [hunk 1]
```diff
@@ -551,21 +551,21 @@ def decode(a, encoding=None, errors=None):
...
 
     Notes
     -----
     The type of the result will depend on the encoding specified.
 
     Examples
     --------
-    >>> c = np.array(['aAaAaA', '  aA  ', 'abBABba'])
+    >>> c = np.array([b'\x81\xc1\x81\xc1\x81\xc1', b'@@\x81\xc1@@',
+                      b'\x81\x82\xc2\xc1\xc2\x82\x81'])
```
Comment (WarrenWeckesser) [inline numpy/core/defchararray.py:563]: In an interactive Python shell, when a line is continued, the prompt becomes `... `:
```suggestion
...                   b'\x81\x82\xc2\xc1\xc2\x82\x81'])
```

File: numpy/core/defchararray.py [hunk 2]
```diff
@@ -551,21 +551,21 @@ def decode(a, encoding=None, errors=None):
...
     >>> c
+    array([b'\x81\xc1\x81\xc1\x81\xc1', b'@@\x81\xc1@@',
+           b'\x81\x82\xc2\xc1\xc2\x82\x81'], dtype='|S7')
```
Comment (WarrenWeckesser) [inline numpy/core/defchararray.py:566]:
```suggestion
...        b'\x81\x82\xc2\xc1\xc2\x82\x81'], dtype='|S7')
```

**Review by WarrenWeckesser — CHANGES_REQUESTED** [review]
This needs a few small tweaks, otherwise looks good.

**Comment by BrentTanKian** [comment]
Hey @WarrenWeckesser, I'll take a look and put in these changes some time next week. Will come back to you if I have more questions :)

**Comment by mattip** [comment]
@BrentTanKian ping

**Comment by BrentTanKian** [comment]
Hey @mattip, my apologies. Got bogged down by some stuff the last couple of weeks. I'll look to put in the new changes by the end of the week

**Comment by BrentTanKian** [comment]
Hi @mattip, @WarrenWeckesser, I may have lost the branch/fork that contained all the changes mentioned here, and have created a new pull request (#22309) to factor in all the new code/changes you have mentioned here

**Comment by seberg** [comment]
Closing, since it is continued in gh-22309.  The push by Matti probably confused you, since you would have had to pull Matti's changes first to continue working on it.

**Pull request closed without merging**